ignite/handlers/checkpoint.py

```diff
@@ -350,19 +350,24 @@ def __call__(self, engine: Engine) -> None:
             }
             filename = filename_pattern.format(**filename_dict)
 
-            if any(item.filename == filename for item in self._saved):
-                return
-
             metadata = {
                 "basename": "{}{}{}".format(self.filename_prefix, "_" * int(len(self.filename_prefix) > 0), name),
                 "score_name": self.score_name,
                 "priority": priority,
             }
 
-            if not self._check_lt_n_saved():
-                item = self._saved.pop(0)
+            filename_already_exists = any(item.filename == filename for item in self._saved)
+
+            if filename_already_exists:
                 if isinstance(self.save_handler, BaseSaveHandler):
-                    self.save_handler.remove(item.filename)
+                    self.save_handler.remove(filename)
+                # list is purged
+                self._saved = [item for item in self._saved if item.filename != filename]
```

Comment:
I'd like to simplify `any(item.filename == filename for item in self._saved)` and `self._saved = [item for item in self._saved if item.filename != filename]` somehow.

Reply:
Ok I think about it.